java/arcs/core/data/expression/PaxelParser.kt

```diff
@@ -204,15 +208,25 @@ object PaxelParser {
             )
         }
 
+    private val orderBySelectors =
+        (refinementExpression + many(comma + refinementExpression)).map { (o, r) ->
+            listOf(o) + r
+        }
+
+    private val orderByExpression = -token("orderby") +
+        (whitespace + orderBySelectors + optional(ows + token("descending"))).map {
+                (selectors, descending) ->
+            Expression.OrderByExpression<Any>(
+            selectors[0] as Expression<Sequence<kotlin.Unit>>,
```

Comment:
I love those hacks we do here 🤣 I spend 2-3 minutes figuring out what this is doing and then remember what this is...

Reply:
Sorry, yeah, I'm gonna remove this hack in the near future after I craft a new combinator for it. We should be able to write something like 

```
val parser = foo qualifiedBy bar
```

which would mean that the `invoke()` method would invoke `foo.invoke()`, but `foo.invoke' would first invoke 'bar.invoke', get the result (e.g. the QualifiedExpression), then foo.invoke()'s 'success' branch would pass in a Pair<QualifiedExpr, OtherArgs> to it's .map {} so it could construct the node with an immutable val. 

But I need to think on it a little more.